utils/s2n_random.c

```diff
@@ -310,6 +310,7 @@ S2N_RESULT s2n_rand_cleanup_thread(void)
 {
     GUARD_AS_RESULT(s2n_drbg_wipe(&per_thread_private_drbg));
     GUARD_AS_RESULT(s2n_drbg_wipe(&per_thread_public_drbg));
+    zero_if_forked = 0;
```

Comment:
From the pr description, it looks like we should only call `s2n_cleanup` when the thread is actually destroyed by the thread pool. This change will force drbg to be initialized multiple times in a thread [code](https://github.com/awslabs/s2n/blob/ecbf2b17e74a6b49b48dc261c91e2670f4819838/utils/s2n_random.c#L92-L93). I am not sure this is the fix we need.

Reply:
Yes, the idea is to clean up and reinitialize resources in server frameworks where thread pools and thread managements (spinning up threads and terminating idle workers) are hidden from end users.